### ISSUE TITLE
"Smart" Previous

### ISSUE THREAD

**Issue by TechnoSparks** [gnumdk/lollypop#1077]
**_This is an enhancement request_**

Greetings

I am not so sure of what this feature is called, but I recall reading it somewhere that it is called “smart previous”.

Basically, clicking on the Previous button places the seek to the start of the song instead of playing the previous song. The next click (in a predefined milliseconds) will play the previous song. This basically acts as a quick repeat for the same song (at least, how i use it). This behaviour exists on Gnome Music by default.

I believe that this may not be the interesting feature for the developer. An additional **option** to allow this behaviour would be extremely appreciated and be excited upon.

All in all, in the end, the choice ends in the hands of the developer.

Warm regards

**Comment by gnumdk** [comment]
What the difference with: "clicking progress bar at start" ?

**Comment by TechnoSparks** [comment]
The difference comes when you are using media keys or any other forms of controls shortcut, e.g. the one in the notification area of Gnome. Lollypop is very beautiful, but most of the time, opening the software multiple times to do simple something can be cumbersome and inefficient. Hence the use of shortcut keys and whatnot.

I am also aware that semantically “previous” means “previous”, hence why I suggested that this requested feature should be an option that is disabled by default

**Comment by gnumdk** [comment]
I will not add an option, so the question is about making this default.

**Comment by TechnoSparks** [comment]
Well, in that case, the only reason I could think of if making this feature by default is because of _familiarity_ 😅 (not sure if this word would best describe the meaning i am trying to convey)

Most media players today have this feature by default. Windows' Groove, RhythmBox, Banshee, Windows Media Player and many Android music players (with notable exception goes to Poweramp which disables this by default). MP3 players also have this feature by default as well and undeniably all Walkman players.. Most people may be familiar with this method given of how abundant this feature existed. But of course, some may hate it as well (look, i have a thumbs down on this thread's main post) hence the reason i am requesting it as an option.

Seriously though, with all respect, the choice is yours to implement it in this wonderful player. 😊

**Comment by gnumdk** [comment]
No, for such feature, I prefer giving the choice to users, I really never use backward button ;)

**Comment by gnumdk** [comment]
https://framadate.org/yWhixFyFvJOFa7te

**Comment by M5oul** [comment]
Wow, my vote have been changed.

**Comment by gnumdk** [comment]
hmm, it's a dividing feature :)

**Comment by gnumdk** [comment]
Ok, you win ;) Just tested, it's a 4 line patch so will add this as a dconf option.

**Comment by gnumdk** [comment]
You need to enable feature in dconf

**Comment by TechnoSparks** [comment]
Sorry for the long hiatus, apparently my usual online-habits has a disruption 😅

I have tested the implemented feature, and it works the way it is. I would like to humbly thank you for the implementation of my request.

Now Lollypop feels complete. Time to remove the other music player 😄

**Comment by Pointedstick** [comment]
For future reference: the exact command to enable this is `gsettings set org.gnome.Lollypop smart-previous true`

**Comment by fbruetting** [comment]
Not available anymore 😞

**Comment by gnumdk** [comment]
? Here it works.

**Comment by gnumdk** [comment]
@fbruetting 
https://github.com/gnumdk/lollypop/blob/master/data/org.gnome.Lollypop.gschema.xml#L264
https://github.com/gnumdk/lollypop/blob/master/lollypop/player.py#L78

**Comment by fbruetting** [comment]
```
$ gsettings get org.gnome.Lollypop smart-previous
No such schema “org.gnome.Lollypop”
```

Maybe because of Flatpak…?

**Comment by gnumdk** [comment]
Yes

**Comment by gnumdk** [comment]
flatpak run --command=sh org.gnome.Lollypop

**Comment by fbruetting** [comment]
I haven’t find anything useful, so many thanks – works!
